common/utils/iiif.js

```diff
@@ -7,6 +7,23 @@ import {
   type IIIFMediaElement,
 } from '../model/iiif';
 
+export function getServiceId(currentCanvas: ?IIIFCanvas) {
+  if (!currentCanvas) return null;
+  const serviceSrc = currentCanvas.images[0].resource.service;
+  if (serviceSrc) {
+    if (Array.isArray(serviceSrc)) {
+      const service = serviceSrc.find(
+        item => item['@context'] === 'http://iiif.io/api/image/2/context.json'
+      );
+      return service && service['@id'];
+    } else {
+      return serviceSrc['@id'];
+    }
+  } else {
+    return null;
+  }
+}
```

Comment:
Might be worth seeing if we can get the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) in for this stuff?